tests/integration/testlib/vxlan.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import libnmstate
+
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import VXLAN
+
+
+def vxlans_up(vxlans):
+    vxlans_state = []
+    for vxlan in vxlans:
+        vxlans_state.append(vxlan_up(*vxlan))
+    return {Interface.KEY: vxlans_state}
+
+
+def vxlans_absent(vxlans):
+    vxlans_state = []
+    for vxlan in vxlans:
+        vxlans_state.append(vxlan_absent(*vxlan))
+    return {Interface.KEY: vxlans_state}
+
+
+@contextmanager
+def apply_state(setup_state, teardown_state):
+    try:
+        libnmstate.apply(setup_state)
```

Comment:
I think this should be outside the `try` block. If `apply` fails, it should clean-up itself.